5. You have a dataset named customer_data.csv that contains information about customers. The dataset has columns customer_id, age, gender, email, and purchases. Write a Python script to read the dataset, perform data cleaning by removing duplicates and handling missing values, and then calculate the average number of purchases for each gender.

In [1]:
#Import the libraries
import pandas as pd
import numpy as np
import requests
import matplotlib.pyplot as plt
import io

In [2]:
#Download and load the dataset
file_id = '17YBlvt8l0S0WQWEhpW7DsuXATe3plgFB'
url = f'https://drive.google.com/uc?id={file_id}'
download = requests.get(url).content
# Read the CSV file into a Pandas DataFrame
df = pd.read_csv(io.StringIO(download.decode('utf-8')))

In [3]:
#View the data
df.head(10)

customer_id  age gender            email  purchases
0         1001   35      M  abc@example.com       10.0
1         1002   28      F  def@example.com        5.0
2         1003   40      F  ghi@example.com        NaN
3         1004   45      M  jkl@example.com        8.0
4         1005   33      M  mno@example.com       15.0
5         1006   27      F  pqr@example.com        NaN
6         1007   50      M  stu@example.com       12.0
7         1008   29      F  vwx@example.com        7.0
8         1009   38      F  yza@example.com        9.0

In [4]:
#View the shape of the dataframe
df.shape

(9, 5)

In [6]:
#Check the NULL values
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9 entries, 0 to 8
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   customer_id  9 non-null      int64  
 1   age          9 non-null      int64  
 2   gender       9 non-null      object 
 3   email        9 non-null      object 
 4   purchases    7 non-null      float64
dtypes: float64(1), int64(2), object(2)
memory usage: 488.0+ bytes


There are 2 NULL values in the purchases.Lets see the summary of the data

In [7]:
df.describe()

customer_id        age  purchases
count     9.000000   9.000000   7.000000
mean   1005.000000  36.111111   9.428571
std       2.738613   7.912718   3.309438
min    1001.000000  27.000000   5.000000
25%    1003.000000  29.000000   7.500000
50%    1005.000000  35.000000   9.000000
75%    1007.000000  40.000000  11.000000
max    1009.000000  50.000000  15.000000

The range of the purchases is from 5 to 11 and the mean of the purchases is 9.4 and the median is 9 so I can use mean to fill the missing values.

In [8]:
#Finding the mean of the column purchases
mean_value=df['purchases'].mean()

In [9]:
# Replace NaNs in column purchases with the
# mean of values in the same column
df['purchases'].fillna(value=mean_value, inplace=True)

In [10]:
#Check the NULL values
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9 entries, 0 to 8
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   customer_id  9 non-null      int64  
 1   age          9 non-null      int64  
 2   gender       9 non-null      object 
 3   email        9 non-null      object 
 4   purchases    9 non-null      float64
dtypes: float64(1), int64(2), object(2)
memory usage: 488.0+ bytes


In [11]:
df.groupby('gender')['purchases'].mean()

gender
F     7.971429
M    11.250000
Name: purchases, dtype: float64

We could see the average purchases of gender Female is 7.97 and for gender Male is 11.25. Male is purchased more than Female